src/main/java/de/csbdresden/csbdeep/io/OutputProcessor.java

```diff
@@ -5,15 +5,11 @@
 
 import de.csbdresden.csbdeep.network.model.ImageTensor;
 import de.csbdresden.csbdeep.task.Task;
-import net.imagej.Dataset;
-import net.imagej.DatasetService;
 import net.imglib2.RandomAccessibleInterval;
 import net.imglib2.type.numeric.RealType;
 
 public interface OutputProcessor<T extends RealType<T>> extends Task {
```

Comment:
You could type the `OutputProcessor` on the output type to prevent the unsafe casts of the output:
```java
public interface OutputProcessor<T extends RealType<T>, O> extends Task {
    O run(...);
}
```

Reply:
Good idea!